utils/data.py

```diff
@@ -33,57 +40,68 @@ def __init__(self, calendar, prices, sales, seq_len=1, horizon=1):
         super(Dataset, self).__init__()
 
         # get data constant per store-item in one array
-        self.day = np.concatenate((
+        self.day = (
             self._weekdays(calendar),
             self._weeks(calendar),
             self._monthdays(calendar),
             self._months(calendar),
             self._years(calendar),
-            self._event_types(calendar)),
-            axis=1
-        ).astype(np.float32)
-
-        # sort sales table for consistency
-        sales = sales.sort_values(by=['store_id', 'item_id'])
-        sales.index = range(sales.shape[0])
+            self._event_types(calendar)
+        )
+        self.day = tuple(map(self._normalize, self.day))
+        self.day = np.concatenate(self.day, axis=1).astype(np.float32)
 
         # get data different per store-item
         self.snap = self._snap(calendar, sales)
         self.prices = self._sell_prices(calendar, prices)
         self.sales = self._unit_sales(sales)
 
+        ForecastDataset.num_const = self.day.shape[1]
+        ForecastDataset.num_var = 3
+        ForecastDataset.num_groups = self.sales.shape[1]
         self.seq_len = seq_len
         self.horizon = horizon
 
+    @staticmethod
+    def _normalize(input):
+        """Normalizes one-hot vectors to z-scores."""
+        num_classes = input.shape[1]
+
+        mean = 1 / num_classes
+        std = math.sqrt((num_classes - 1) * mean**2)
+
+        return (input - mean) / std
```

Comment:
Ik vind het nog steeds een beetje vreemd om one hot vectors the normaliseren omdat het eigenlijk gewoon meerdere binary variables zijn. Ik heb even kort gezocht op internet en het lijkt niet gebruikelijk, heb je zelf wel eerder gezien dat one hot vectors worden genormaliseerd?

Reply:
There isn't really a consensus, basically: "do what works best". The next run will be without normalizing to see what works better